File: february_2023/ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        j = len(height) - 1
        areas = []
        while i != j:
            area = min(height[i], height[j]) * (j - i)
            areas.append(area)
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
        return max(areas)

File: february_2023/ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length - 1;
        List<Integer> areas = new ArrayList<>();
        int counter = 0;
        while (i != j){
            int area = Math.min(height[i], height[j]) * (j - i);
            areas.add(area);
            if (height[i] > height[j]){
                j -= 1;
            }
            else {
                i += 1;
            }
            counter += 1;
        }
        int max = 0;
        for (int num : areas){
            if (num > max){
                max = num;
            }
        }
        return max;
    }
}
